### ISSUE TITLE
Ctrl+C is blocked by TerminalGUI when run on mono

### ISSUE THREAD

**Issue by adienakhmad** [migueldeicaza/gui.cs#218]
Hi, it seems like `Ctrl-C` (`CancelPressKey` event/`SIGINT`) is blocked in when running on Mono. Is there a way to close TerminalGUI on this key event?

What I've done so far

1. Subscribe` Console.CancelKeyPress` > works on window
2. Listen for `UnixSignal `using `Mono.Posix` > does not work

I launched a separate task specifically to wait for unix signal,
```c#

Log.LogInfo( "Listening for Unix Signal.." );
      System.Threading.Tasks.Task.Run( () =>
                                       {
                                         UnixSignal.WaitAny( new[]
                                                             {
                                                               new UnixSignal( Signum.SIGTERM ),
                                                               new UnixSignal( Signum.SIGQUIT ),
                                                               new UnixSignal( Signum.SIGHUP ),
                                                               new UnixSignal( Signum.SIGINT )
                                                             } );

                                         Log.LogInfo( "Received unix signal, stopping terminal GUI.." );
                                         StopTerminalGui();
                                       } );
```

and here's the content of `StopTerminalGui`

```c#
private static void StopTerminalGui()
    {

      // terminate Terminal GUI so Main() can continue to shutdown/clean up
      Application.RequestStop();
    }
```

**Comment by adienakhmad** [comment]
My bad, I should have just inherit `TopLevel` and override `ProcessKey` event from there
